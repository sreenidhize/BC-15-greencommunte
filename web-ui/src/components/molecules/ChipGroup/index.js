import React, { useEffect, useState } from "react";
import { makeStyles } from "@material-ui/core/styles";
import Chip from "../../atoms/Chip";
import { Grid, Typography } from "@material-ui/core";
import { connect } from "react-redux";
import { apis } from "../../../utils/resources/resources";
import {
  chipAndFilterGroupDispatch,
  experienceLevel,
  jobType,
  landingPageLocationAndSkillParams,
  // landingPageLocationAndSkillParams,
  maxDistance,
} from "../../../utils/utils/utils";
import { convertServiceToClientJobs } from "../../../state/converter/jobConverter";
import { points } from "../../../utils/constants/constant";

const useStyles = makeStyles((theme) => ({
  root: {
    paddingTop: theme.spacing(1),
    paddingBottom: theme.spacing(1),
  },
  chips: {
    display: "flex",
    flexWrap: "wrap",
    "& > *": {
      margin: theme.spacing(1, 2),
    },
  },
  clearAll: {
    marginTop: theme.spacing(2),
    marginLeft: 50,
    color: theme.palette.secondary.main,
    height: 1,
    textAlign: "center",
    cursor: "pointer",
  },
}));

const ChipGroup = (props) => {
  const classes = useStyles();
  const { dataTestId, Filters } = props;
  const getChips = (objectArray) => {
    const chips = [];
    for (let i = 0; i < objectArray.length; i++) {
      if (objectArray[i].checked === true) {
        chips.push(objectArray[i].label);
      }
    }
    return chips;
  };
  const [localFilters, setLocalFilters] = useState(Filters);
  const [chipState, setChipState] = useState(getChips(localFilters));
  useEffect(() => {
    setChipState(getChips(localFilters));
    setLocalFilters(localFilters);
    // eslint-disable-next-line
  }, [Filters]);
  let params = {};
  const initialJobLocation = [];
  const finalJobLocation = "";
  params = landingPageLocationAndSkillParams(
    params,
    initialJobLocation,
    finalJobLocation,
    props.jobLocation,
    props.skill,
    points
  );
  console.log(params);
  const fetchJobsUrl = new URL(apis.GET_JOBS);
  const handleSearch = () => {
    Object.keys(params).forEach((key) =>
      fetchJobsUrl.searchParams.append(key, params[key])
    );
    fetch(fetchJobsUrl)
      .then((res) => res.json())
      .then((res) => props.fetchJobs(convertServiceToClientJobs(res)))
      .catch((e) => console.log(e));
  };

  const handleDelete = (chipToDelete) => () => {
    const chips = [...chipState];
    const index = chips.indexOf(chipToDelete);
    if (index !== -1) {
      chips.splice(index, 1);
      setChipState(chips);
    }
    let flag = 0;
    for (let i = 0; i < localFilters.length; i++) {
      if (localFilters[i].label === chipToDelete) {
        localFilters[i].checked = false;
      }
      if (localFilters[i].label === "Yes") {
        flag = 1;
      }
    }
    props.getFilters(localFilters);
    setLocalFilters(localFilters);
    if (chipToDelete === "Yes" || flag === 1) {
      params = { ...params, is_green_commute: false };
    } else if (flag === 0) {
      params = { ...params, is_green_commute: true };
    }
    maxDistance(localFilters)
      ? (params = {
          ...params,
          distance: maxDistance(localFilters),
          is_green_commute: props.filters[4].checked,
        })
      : (params = { ...params, is_green_commute: props.filters[4].checked });
    jobType(props.filters)
      ? (params = {
          ...params,
          job_type: jobType(props.filters),
          is_green_commute: props.filters[4].checked,
        })
      : (params = { ...params, is_green_commute: props.filters[4].checked });
    experienceLevel(props.filters)
      ? (params = {
          ...params,
          experience_level: experienceLevel(props.filters),
          is_green_commute: props.filters[4].checked,
        })
      : (params = { ...params, is_green_commute: props.filters[4].checked });
    props.skill[0]
      ? (params = {
          ...params,
          skill: props.skill[0],
          is_green_commute: props.filters[4].checked,
        })
      : (params = { ...params, is_green_commute: props.filters[4].checked });
    handleSearch();
  };
  const clearChips = () => {
    for (let i = 0; i < localFilters.length; i++) {
      localFilters[i].checked = false;
    }
    props.getFilters(localFilters);
    setChipState([]);
  };
  const clearAllChips = () => {
    clearChips();
    props.skill[0]
      ? (params = {
          ...params,
          skill: props.skill[0],
          is_green_commute: false,
        })
      : (params = { ...params, is_green_commute: false });
    handleSearch();
  };
  // eslint-disable-next-line
  const Chips = localFilters.map((object) => {
    if (object.checked === true) {
      if (object.label !== "Yes") {
        return (
          <Chip
            content={object.label}
            handleDelete={handleDelete(object.label)}
          ></Chip>
        );
      } else {
        return (
          <Chip
            content={"Green Commute"}
            handleDelete={handleDelete(object.label)}
          ></Chip>
        );
      }
    }
  });
  return (
    <div data-testid={dataTestId} className={classes.root}>
      <Grid container>
        <Grid item className={classes.chips} xs={10}>
          {Chips}
        </Grid>
        {chipState.length !== 0 && (
          <Grid item xs={2}>
            <Typography onClick={clearAllChips} className={classes.clearAll}>
              clear all
            </Typography>
          </Grid>
        )}
      </Grid>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    Filters: state.filters,
    jobLocation: state.jobLocation,
    userLocation: state.userLocation,
    skill: state.skills,
    filters: state.filters,
    draftJob: state.draftJob,
    cabOptions: state.cabOptions,
    metroOptions: state.metroOptions,
  };
};

chipAndFilterGroupDispatch();

export default connect(mapStateToProps, chipAndFilterGroupDispatch)(ChipGroup);
