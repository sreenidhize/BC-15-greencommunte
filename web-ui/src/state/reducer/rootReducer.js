import {
  JOBS_RECEIVED,
  ONCHANGE_SKILLS,
  SAVEDJOBS_RECEIVED,
  ONCHANGE_ALLSKILLS,
  ONCHANGE_JOBLOCATION,
  ONCHANGE_USERLOCATION,
  JOB_RECEIVED,
  GET_FILTERS,
} from "../action/actionTypes";

const initState = {
  jobs: [],
  searchOptions2: [
    "Hyderabad",
    "Indore",
    "Mumbai",
    "Patna",
    "Banglore",
    "Delhi",
  ],
  searchOptions1: [],
  userLocation: "",
  jobLocation: "",
  skills: [],
  draftJob: {},
  savedJobs: [],
  cabOptions: [
    {
      type: "OLA",
      value: "40",
      url:
        "https://res-5.cloudinary.com/crunchbase-production/image/upload/c_lpad,h_256,w_256,f_auto,q_auto:eco/v1459446250/xx0u5xwwzygivm67ogsk.jpg",
    },
    {
      type: "UBER",
      value: "40",
      url:
        "https://semaine-production.s3.amazonaws.com/products/product/image/6827/large_Uber-app.jpg",
    },
    {
      type: "Rapido",
      value: "40",
      url:
        "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSWOLJJfUAbhh1JVtRLgNCexBH9_cC2L3NvqA&usqp=CAU",
    },
  ],
  metroOptions: [
    {
      type: "bus",
      fare: "15$",
      time: "20 mins",
      description: "Catch a bus 38X at 3.42 PM to Secunderabad bus stand.",
      lateStatus: "2 mins",
      isLate: true,
    },
    {
      type: "walk",
      time: "5 mins",
      description: "Head north on St. Johns road.",
    },
    {
      type: "metro",
      fare: "56$",
      time: "38 mins",
      description: "Catch a blue line metro towards Raidurg.",
      isLate: false,
    },
  ],
  filters: [
    { label: "0-10 kms", checked: false },
    { label: "10-20 kms", checked: false },
    { label: "20-30 kms", checked: false },
    { label: "30-40 kms", checked: false },
    { label: "Yes", checked: false },
    { label: "Past 24 hours", checked: false },
    { label: "Past week", checked: false },
    { label: "Past month", checked: false },
    { label: "Anytime", checked: false },
    { label: "Full Time", checked: false },
    { label: "Internship", checked: false },
    { label: "Contract", checked: false },
    { label: "Remote", checked: false },
    { label: "Fresher", checked: false },
    { label: "Mid-level", checked: false },
    { label: "Director", checked: false },
    { label: "Executive", checked: false },
    { label: "Metro", checked: false },
    { label: "Motor cycle", checked: false },
    { label: "Bus", checked: false },
    { label: "Car Pooling", checked: false },
  ],
};

const rootReducer = (state = initState, action) => {
  switch (action.type) {
    case JOBS_RECEIVED:
      return {
        ...state,
        jobs: action.payload,
      };
    case ONCHANGE_JOBLOCATION:
      return {
        ...state,
        jobLocation: action.payload,
      };
    case ONCHANGE_USERLOCATION:
      return {
        ...state,
        userLocation: action.payload,
      };
    case ONCHANGE_SKILLS:
      return {
        ...state,
        skills: action.payload,
      };
    case ONCHANGE_ALLSKILLS:
      return {
        ...state,
        searchOptions1: action.payload,
      };
    case JOB_RECEIVED:
      return {
        ...state,
        draftJob: action.payload,
      };
    case SAVEDJOBS_RECEIVED:
      return {
        ...state,
        savedJobs: action.payload,
      };
    case GET_FILTERS:
      return {
        ...state,
        filters: action.payload,
      };
    default:
      return state;
  }
};

export default rootReducer;
