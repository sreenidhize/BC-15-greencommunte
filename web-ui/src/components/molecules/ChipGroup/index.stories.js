import React from "react";
import ChipGroup from ".";

export default {
  title: "molecules/ChipGroup",
  component: ChipGroup,
};

const Template = (args) => <div style={{width: 690}}><ChipGroup {...args} /></div>;

const chips = ["10-20 km", "Past 24 hours", "Green Commute", "Fresher"];

const clearAll = () => {
  alert("all chips is removed");
};

export const Primary = Template.bind({});
Primary.args = {
  chips: chips,
  clearAll: clearAll,
};