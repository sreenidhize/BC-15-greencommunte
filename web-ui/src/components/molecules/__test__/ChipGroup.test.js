import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import ChipGroup from "../ChipGroup";
import { Provider } from 'react-redux';
import { createStore } from "redux";
import rootReducer from "../../../state/reducer/rootReducer";

const store = createStore(rootReducer);

const testId = "ChipGroup";
const chips = ["10-20 km", "Past 24 hours", "Green Commute", "Fresher"];

global.fetch = jest.fn(() =>
      Promise.resolve()
    );

describe("Chip Group test", () => {
  it("Matches the snapshot", () => {
    const { container } = render(
      <Provider store={store}><ChipGroup chips={chips} dataTestId={testId} /></Provider>
    );
    expect(container).toMatchSnapshot();
  });

  test("test ChipGroup render", () => {
    const { getByTestId } = render(<Provider store={store}><ChipGroup chips={chips} dataTestId={testId} /></Provider>);
    expect(getByTestId("ChipGroup")).toBeInTheDocument();
  });
});