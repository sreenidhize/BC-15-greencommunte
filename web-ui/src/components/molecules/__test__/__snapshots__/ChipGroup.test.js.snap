// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip Group test Matches the snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
    data-testid="ChipGroup"
  >
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root makeStyles-chips-2 MuiGrid-item MuiGrid-grid-xs-10"
      />
    </div>
  </div>
</div>
`;
